attribute-utils/src/main/java/org/hypertrace/attribute/span/SpanTagResolver.java

```diff
@@ -0,0 +1,86 @@
+package org.hypertrace.attribute.span;
+
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.attribute.Constants;
+import org.hypertrace.core.datamodel.AttributeValue;
+import org.hypertrace.core.datamodel.Event;
+import org.hypertrace.core.span.constants.RawSpanConstants;
+import org.hypertrace.core.span.constants.v1.Envoy;
+import org.hypertrace.core.span.constants.v1.OCAttribute;
+import org.hypertrace.core.span.constants.v1.OCSpanKind;
+import org.hypertrace.core.span.constants.v1.SpanNamePrefix;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SpanTagResolver {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SpanTagResolver.class);
+
+  private static final String OTEL_SPAN_KIND = "span_kind";
+  private static final String OTEL_SPAN_KIND_SERVER_VALUE = "server";
+  private static final String OTEL_SPAN_KIND_CLIENT_CLIENT = "client";
+
+  private static final String OC_SPAN_KIND = Constants.getRawSpanConstant(OCAttribute.OC_ATTRIBUTE_SPAN_KIND);
+  private static final String OC_SPAN_KIND_SERVER_VALUE = Constants.getRawSpanConstant(OCSpanKind.OC_SPAN_KIND_SERVER);
+  private static final String OC_SPAN_KIND_CLIENT_VALUE = Constants.getRawSpanConstant(OCSpanKind.OC_SPAN_KIND_CLIENT);
+  // Spans that are generated by Go Opencensus grpc/http clients have this attribute.
+  private static final String OC_CLIENT_KEY = Constants.getRawSpanConstant(OCAttribute.OC_ATTRIBUTE_CLIENT_KEY);
+
+  // todo: can't find any envoy related attributes in OTEL
+  private static final String envoyOperationNameAttr = RawSpanConstants.getValue(Envoy.ENVOY_OPERATION_NAME);
+  private static final String envoyIngressSpanValue = RawSpanConstants.getValue(Envoy.ENVOY_INGRESS_SPAN);
+  private static final String envoyEgressSpanValue = RawSpanConstants.getValue(Envoy.ENVOY_EGRESS_SPAN);
+
+  public static Boolean checkForEntrySpan(Event event) {
+    Map<String, AttributeValue> attributeMap = event.getAttributes().getAttributeMap();
+    // Figure out if event is entry or exit based on other span attributes
+    Boolean isEntry = null;
+    if (attributeMap.containsKey(OTEL_SPAN_KIND)) {
```

Comment:
nit: could we simplify this method? There are too many if statements

Reply:
I will simplify it. I am actually breaking this pr into smaller pr. Will address it there.